### PR TITLE
#89 wallet integration test

### PR DESCRIPTION
**Pull request by WeiyuSun**
#### /wallet GET: 

1. get wallets with no params, just with default `limit`
2. get wallet by wallet name
3. try to get a wallet which from other wallet account
4. test `offset`(and `limit`)
5. get wallet by uuid
#### /wallet POST: 

1. wallet name validation(length, characters)

### CODE REVIEW

File: __tests__/wallet-post.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        const resB = await request(server)
+            .post('/wallets')
+            .send({wallet: '2023.7.26CodingAtCanadaNiceToMeetYouuuuuuuuuuuuuuuu'})
+            .set('treetracker-api-key', apiKey)
+            .set('content-type', 'application/json')
+            .set('Authorization', `Bearer ${bearerTokenA}`);
+
+        expect(resB).property('statusCode').to.eq(422);
+    })
```
Comment (dadiorchen) [inline __tests__/wallet-post.spec.js:75]: If we already cover this edge case in unit test, then I think we don't need to do this in int test, I feel unit test is a better choice to cover these edge case:
1. It is less expensive, 
2. It is easier to build the test case.

**Review by dadiorchen — APPROVED** [review]
